forc/src/ops/forc_template.rs

```diff
@@ -0,0 +1,169 @@
+use crate::cli::TemplateCommand;
+use crate::utils::{defaults, SWAY_GIT_TAG};
+use anyhow::{anyhow, Context, Result};
+use forc_pkg::{
+    fetch_git, fetch_id, find_dir_within, git_commit_path, pin_git, Manifest, SourceGit,
+};
+use forc_util::validate_name;
+use fs_extra::dir::{copy, CopyOptions};
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+use std::{env, fs};
+use sway_utils::constants;
+use url::Url;
+
+pub fn init(command: TemplateCommand) -> Result<()> {
+    validate_name(&command.project_name, "project name")?;
+    let template_name = match command.template_name.clone() {
```

Comment:
```suggestion
    // The name used for the temporary local repo directory used for fetching the template.
    let local_repo_name = match command.template_name.clone() {
```
Perhaps something like this is a better name for this binding as it's only used for the local git repository, and isn't necessarily the name of the template project.